added challenge: Lending Challenge on Aave

Turupawn opened the pull request.

emilyJLin95 commented on src/challenges/defi-challenges/simple-lending.mdx:
```diff
@@ -0,0 +1,221 @@
+index: 3
+level: 3
+website: "https://github.com/LevelUpWeb3/SimpleTokenSwap-Challenge"
+summary: Introduction to implementing a simple token swap contract by understanding Uniswap swap function.
```
I think we need to change this metadata and GH repo?

Turupawn commented:
@emilyJLin95 @PrintRH I updated the challenge by making some corrections and adding an explanation on how to run unit the test. I feel like this is ready to be merged, please help me reviewing :pray:

emilyJLin95 reviewed (commented):
For the most part LGTM - do you have permissions to create the challenge repo under the levelupweb3 org?

PrintRH commented:
@emilyJLin95 - I am not sure, but I don't think so. To keep things moving quicker, I have created the challenge repo here: https://github.com/LevelUpWeb3/SimpleLending-Challenge

Turupawn commented:
@PrintRH Thanks for creating the repo, I added the code there and updated the challenge. Also run the `processContentMarkdown.js` and manually fixed the expected conflicts

PrintRH commented on src/challenges/defi-challenges/simple-lending.mdx:
```diff
@@ -0,0 +1,244 @@
+---
+name: Simple Lending
+index: 3
```
Can we update this to index "4"? So, this follows after recent challenge from 0x.

PrintRH commented on src/challenges/defi-challenges/simple-lending.mdx:
```diff
@@ -0,0 +1,244 @@
+labels: ["DeFi"]
+---
+
+import ContentCopy from "src/components/ContentCopy";
```
In this new update, ContentCopy integration is no longer required. You can directly write code in traditional markdown format.

PrintRH commented on src/challenges/defi-challenges/simple-lending.mdx:
```diff
@@ -0,0 +1,244 @@
+
+If you get stuck, feel free to ask for help in [Level Up Telegram group](https://t.me/+PdNbk5milo1mMTAy).
+
+<ContentCopy code={`// SPDX-License-Identifier: MIT
```
Remove ContentCopy component and can use the standard format writing code in markdown.

PrintRH commented on src/challenges/defi-challenges/simple-lending.mdx:
```diff
@@ -0,0 +1,244 @@
+## Objective
+
+Your task is to:
+1. Develop your own contract with `deposit` and `withdraw` functionalities based on the provided base contract.
```
Challenge below seems to define `stake` and `withdraw` function. Should we update `deposit` to `stake`?

Also, I notice that the majority of the challenge calls out `supply` function. Does it make more sense to just use `supply` for rest of the challenge?

Turupawn replied:
The thing is the Aave functions are called `deposit` and `withdraw`, and I named the challenge functions `stake` and `withdraw` so they are different. But yeah that sounds confusing.

So I think we should rename the challenges functions to `proxyDeposit` and `proxyWithdraw` to make a clear distinction. Other option could be `stake` and `unstake`. What do you think?

PrintRH replied:
That makes sense! Personally, I think `stake` and `unstake` provides more clarity. I think then just updating the `deposit` and `withdraw` pair for the rest of this challenge will be sufficient, and we can ship it!

PrintRH reviewed (commented):
@Turupawn - Challenge is coming along great! Left a few comments for fixes then I think we should be ready to ship this!

emilyJLin95 commented:
Oh @Turupawn 

Realized, for removing the existing swaps challenge, could we just add it as part of this PR since this is getting merged and reviewed now?

Turupawn commented:
@PrintRH @emilyJLin95 
I addressed all comments + temporarily removed the lending challenge

PrintRH commented:
@Turupawn Challenges shaping up nicely!

One last thing - can you also run the `build` command as a last push? I am noticing some dev fragments which might break some links/contents in prod.

Turupawn commented:
@PrintRH I just run it, did it work?

PrintRH reviewed (approved):
It worked!

Changes LGTM! Let's ship this. Thank you for your hard work here @Turupawn !

PrintRH commented:
Oh wait one last thing @Turupawn - merging is still blocked, can you pull and merge the latest changes from `main` to this branch?

That should resolve the merge error.

Turupawn commented:
@PrintRH done1
